Gems/Atom/Feature/Common/Code/Source/Stars/StarsFeatureProcessor.cpp

```diff
@@ -0,0 +1,201 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <Stars/StarsFeatureProcessor.h>
+
+#include <Atom/RHI/DrawPacketBuilder.h>
+#include <Atom/RHI.Reflect/InputStreamLayoutBuilder.h>
+
+#include <Atom/RPI.Public/RPIUtils.h>
+#include <Atom/RPI.Public/Shader/Shader.h>
+#include <Atom/RPI.Public/Scene.h>
+#include <Atom/RPI.Public/View.h>
+#include <Atom/RPI.Public/ViewportContext.h>
+#include <Atom/RPI.Public/ViewportContextBus.h>
+
+namespace AZ
+{
+    namespace Render
+    {
+        void StarsFeatureProcessor::Reflect(ReflectContext* context)
+        {
+            if (auto* serializeContext = azrtti_cast<SerializeContext*>(context))
+            {
+                serializeContext
+                    ->Class<StarsFeatureProcessor, FeatureProcessor>()
+                    ->Version(1);
+            }
+        }
+
+        void StarsFeatureProcessor::Activate()
+        {
+            m_sceneSrg = GetParentScene()->GetShaderResourceGroup();
+
+            const char* shaderFilePath = "Shaders/stars/stars.azshader";
+            m_shader = RPI::LoadCriticalShader(shaderFilePath);
+
+            m_drawListTag = m_shader->GetDrawListTag();
+
+            m_starsDataBufferIndex.Reset();
+
+            // do not create the pipeline state here, the scene has no pipelines at this point
+        }
+
+        void StarsFeatureProcessor::Deactivate()
+        {
+            m_sceneSrg = nullptr;
+            m_shader = nullptr;
+        }
+
+        void StarsFeatureProcessor::Simulate(const FeatureProcessor::SimulatePacket& packet)
+        {
+            AZ_PROFILE_SCOPE(RPI, "StarsFeatureProcessor: Simulate");
+            AZ_UNUSED(packet);
+
+            if (!m_enabled)
+            {
+                return;
+            }
+
+            const AZ::RPI::ViewportContextPtr viewportContext = AZ::Interface<AZ::RPI::ViewportContextRequestsInterface>::Get()->GetDefaultViewportContext();
+            const AzFramework::WindowSize windowSize = viewportContext->GetViewportSize();
+            const float vpWidth = static_cast<float>(windowSize.m_width);
+            const float vpHeight = static_cast<float>(windowSize.m_height);
+            const float size = m_radiusFactor * AZStd::min<float>(1.f, AZStd::min<float>(vpWidth / 1280.f, vpHeight / 720.f));
+
+            AZ::ScriptTimePoint timePoint;
+            TickRequestBus::BroadcastResult(timePoint, &TickRequestBus::Events::GetTimeAtCurrentTick);
```

Comment:
nit: We really should add this to the global scene srg if it's not there already....

Reply:
It is! I need to convert the shader to use it instead.